solvers/fista_automatic_restart.py

```diff
@@ -32,47 +32,42 @@ def run(self, callback):
         C = 6.38  # Parameter controlling the restart frequency, 6.38 is the
         # optimal theoretical value (we refer the user to the preprint for
         # more details).
-        inner_iteration = 0
-        n_restarts = 0  # Restart counter
-        last_F = 0
         ite_per_restart = np.array([int(2*C)])
-        F_tab = np.array([self.cost_function(w)])
-        cb = callback(w)
+        objectives = np.array([self.cost_function(w)])
+        should_run = callback(w)
```

Comment:
I introduced the boolean should_run/cb because otherwise the algorithm never stops running (by calling callback(w) directly). I can try again without it.

Reply:
Sorry, I edited my comment, I was suggesting to keep it be stop as soon as the callback is `False`

Reply:
Yes you're right!